packages/datadog-shimmer/src/shimmer.js

```diff
@@ -44,18 +59,19 @@ function wrapFunction (original, wrapper) {
   return wrapped
 }
 
-function wrap (target, name, wrapper) {
-  assertMethod(target, name)
+function wrap (target, name, wrapper, options) {
   if (typeof wrapper !== 'function') {
     throw new Error(wrapper ? 'Target is not a function' : 'No function provided')
   }
 
-  const original = target[name]
-  const wrapped = wrapper(original)
+  let descriptor = Object.getOwnPropertyDescriptor(target, name)
+  const original = descriptor?.get && (!options?.replaceGetter || descriptor.set) ? descriptor.get : target[name]
```

Comment:
I think the logic here is hard to follow and it would benefit with a code comment:

```suggestion
  // Replacing the getter is not supported if the property also has a setter
  const original = descriptor?.get && (!options?.replaceGetter || descriptor.set) ? descriptor.get : target[name]
```